### PR TITLE
Use make-dind, update requested resources and remove default values

### PR DESCRIPTION
**Pull request by inteon**
Based on changes we made in https://github.com/jetstack/testing

In https://github.com/jetstack/testing, we removed the -j limitations. This however has not yet proven to be a great improvement, so we shouldn't upstream that change here yet.

The value `Agent: "kubernetes",` was removed because this is the default prowjob value. An empty value is thus identical to not specifying this value.

### CODE REVIEW

File: prowspecs/specs.go
Comment (SgtCoDFish) [inline prowspecs/specs.go:39]: suggestion: Having older releases have their images be "pinned" was an intentional choice in #90  - does this not defeat the point of that PR?

IMO either we should decide to use CommonTestImage everywhere or leave the older images alone. I'd be happy to use CommonTestImage everywhere, but that was explicitly rejected in #90

File: pkg/prowgen/configurers.go
Comment (SgtCoDFish) [inline pkg/prowgen/configurers.go:30]: question: This change (removing Agent) seems disconnected from the title of the PR. Could you add some detail to the PR description on why this was done?

**Review by SgtCoDFish — CHANGES_REQUESTED** [review]
I think there are some issues here:

1) These changes appear to already be live (https://github.com/jetstack/testing/pull/847, https://github.com/jetstack/testing/pull/844) after having been merged without a review from anyone else on the team :warning: 

2) I think we should be merging changes to config generation either before the changes go live or at the same time as they go live. If I were to regenerate our test config from `master` now it would overwrite a lot of manual changes to the testing config.

3) We're seeing a lot of testgrid failures currently. Notably, some trivy tests have gone stale and stopped reporting results. This could easily be related to these testing changes, but if we want to fix them we're going to drift even further away into a place where cmrel's generated config from the `master` branch doesn't match the testgrid config.

It doesn't seem ideal to me to have changes to our testing config merged without review, and it also defeats the point of having test generation if we merge manual changes to cert-manager testing config.

File: prowspecs/specs.go
Comment (inteon) [reply prowspecs/specs.go:39]: The images were changed from bazelbuild to make-dind in https://github.com/jetstack/testing/pull/809.
The other follow-up PRs just upgraded bazelbuild. Whether we want to keep doing this automatically, I think we have to discuss (eg. the images are still "pinned" to bullseye).

Comment (SgtCoDFish) [reply prowspecs/specs.go:39]: After discussion in standup it seems clear that we want to use `CommonTestImage` for every Image parameter until we have a reason not to :+1:

File: pkg/prowgen/configurers.go
Comment (inteon) [reply pkg/prowgen/configurers.go:30]: done

File: pkg/prowgen/generators.go
Comment (irbekrm) [inline pkg/prowgen/generators.go:154]: FWIW this value is not used for the upgrade test, so there is a diff with the current config in testing where this was manually changed to `700m` https://github.com/jetstack/testing/blob/master/config/jobs/cert-manager/cert-manager/master/cert-manager-master.yaml#L980
If the additional CPU was needed because of the e2e test setup changes where we use more concurrent workers and higher burst/QPS for kube queries, then that's ok as the upgrade test installs cert-manager using a separate install command that doesn't have those opts

Comment (inteon) [reply pkg/prowgen/generators.go:154]: Indeed, the upgrade command does not need 7000m. Also, there might be more tuning necessary in the future.

**Pull request merged**
